### PR TITLE
[A.Zhukevych]Fetch-response

### PR DESCRIPTION
**Pull request by Hollister009**

### CODE REVIEW

File: css-practice/task3/js/fetch-data.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
+/* global fetch, window, document */
+const blogs = 'http://localhost:3000/api/blogs';
```
Comment (dverbovyi) [inline css-practice/task3/js/fetch-data.js]: global variable detected!

File: css-practice/task3/js/fetch-data.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        .then((res) => res.json());
+  }
+
+  return fetchData(url);
+};
+
+const splitResponseData = (res) => {
+  const latestIds = res.latest;
+  const latestItems = res.blogs.filter((item) => latestIds.includes(item.id));
+  const otherItems = res.blogs.filter((item) => !latestIds.includes(item.id));
```
Comment (dverbovyi) [inline css-practice/task3/js/fetch-data.js]: do it in 1 loop

File: css-practice/task3/js/fetch-data.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  renderLatestBlogs(latestItems);
+  renderFooterBlogs(otherItems);
+};
+
+// Parsing json blogs
+const getDataFromObj = (obj) => {
```
Comment (dverbovyi) [inline css-practice/task3/js/fetch-data.js]: not a valid function name
prepareBlogModel is more appropriate

File: css-practice/task3/js/fetch-data.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    return tmpl;
+  }
+};
+
+// Render Footer Blogs
+const renderFooterBlogs = (array) => {
+  const footerBlogs = document.querySelector('.footer__blogs');
+  footerBlogs.innerHTML = '';
+
+  array.forEach((el) => {
```
Comment (dverbovyi) [inline css-practice/task3/js/fetch-data.js]: get rid of this duplication


create a common function

```
function renderBlogs(blogs, container, factory) {
  container.innerHTML = '';
   blogs.forEach(blog => {
    const blog = prepateBlogModel(blog);
    container.append(factory(blog));
  });
}
```
and usage
 
```
renderFooterBlogs(array) {
    const footerBlogs = document.querySelector('.footer__blogs');
    renderBlogs(array, footerBlogs, _createFooterBlogItem)
    ...
}

renderLatestBlogs(array) {
    const latestPost = document.querySelector('.latest-posts');
    renderBlogs(array, latestPost, _createLatestBlogItem)
    ...
}

**Comment by Hollister009** [comment]
I will create a new pull request, cause of unnecessary commits in this one.

**Pull request closed without merging**